registrations/signals.py

```diff
@@ -64,29 +162,26 @@ def signup_group_post_delete(
 
 
 @receiver(
-    pre_save,
-    sender=Event,
-    dispatch_uid="notify_signups_on_event_cancellation_pre_save",
+    post_save,
+    sender=SeatReservationCode,
+    dispatch_uid="seat_reservation_post_save",
 )
-def notify_signups_on_event_cancellation_pre_save(
-    sender: type[Event], instance: Event, **kwargs: dict
+def seat_reservation_post_save(
+    sender: type[SeatReservationCode], instance: SeatReservationCode, **kwargs: dict
 ) -> None:
-    if not (instance.pk and instance.event_status == Event.Status.CANCELLED):
-        return
-
-    old_instance = Event.objects.filter(pk=instance.pk).first()
-    if old_instance and old_instance.event_status != instance.event_status:
-        _send_event_cancellation_notification(instance)
+    transaction.on_commit(
+        lambda: _recalculate_registration_capacities(instance.registration_id)
+    )
 
 
 @receiver(
-    pre_save,
-    sender=RegistrationPriceGroup,
-    dispatch_uid="calculate_registration_price_group_vat_prices",
+    post_delete,
+    sender=SeatReservationCode,
+    dispatch_uid="seat_reservation_post_delete",
 )
-def calculate_registration_price_group_vat_prices(
-    sender: type[RegistrationPriceGroup],
-    instance: RegistrationPriceGroup,
-    **kwargs: dict,
+def seat_reservation_post_delete(
+    sender: type[SeatReservationCode], instance: SeatReservationCode, **kwargs: dict
 ) -> None:
-    instance.calculate_vat_and_price_without_vat()
+    transaction.on_commit(
+        lambda: _recalculate_registration_capacities(instance.registration_id)
+    )
```

Comment:
At the moment registration capacities are updated when seat reservation is created or deleted. How about when seat reservation expires. I think that seats should be released in that case as well.

Reply:
The solution I mentioned in our daily today is now pushed here. I created a new management command for finding and deleting expired reservations. Deleting them will cause the remaining registration capacities to be calculated again. The command can be scheduled in DevOps to be ran every minute